### DISCUSSION TITLE
[Feature] Extra authentication headers for mobile app

### DISCUSSION THREAD

**Discussion by mounte** [immich-app/immich#6671] [Feature Request]
### The feature

If Immich is deployed behind some tunnel or authentication middleware the mobile app no longer works because of not understanding the login flow.
I think a simple addition to the advanced settings would allow the application to work even in these cases with some extra conditions.
If the user can specify an extra http header to be sent with requests many tunnels and middleware solutions can work, as an example for Cloudflare a user can set up a Service Auth token, then if the header contains that token something like
```
CF-Access-Client-Id: <some secret>
```
authentication will work.

So my suggestion is as simple as adding one (or two) configurable text field(s) to advanced setting where user can specify header-name and header-value or if single field the whole string in one go.
This information needs to be stored and appended to all api requests but I think that should be pretty straightforward.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #1604

**Reply by mounte** [reply]
I just realized 😶‍🌫️  sorry for the noise... but thanks for a great project :D
